source/application/AMPIC887DriverServer/AMPIC887HexapodState.cpp

```diff
@@ -0,0 +1,454 @@
+
+		highSoftLimits_.insert(currentAxis, highSoftLimits.value(currentAxis));
+	}
+}
+
+bool AMPIC887HexapodState::softLimitState(AMGCS2::Axis axis) const
+{
+	return softLimitStates_.value(axis, false);
+}
+
+const AMPIC887AxisMap<bool> AMPIC887HexapodState::softLimitStates() const
+{
+	return softLimitStates_;
+}
+
+void AMPIC887HexapodState::setSoftLimitState(AMGCS2::Axis axis, bool softLimitState)
+{
+	if(axis == AMGCS2::UnknownAxis) {
+		return;
+	}
+
+	softLimitStates_.insert(axis, softLimitState);
+}
+
+void AMPIC887HexapodState::setSoftLimitStates(const AMPIC887AxisMap<bool> &softLimitStates)
+{
+	AMPIC887AxisCollection axes = softLimitStates.axes();
+	foreach(AMGCS2::Axis currentAxis, axes) {
+
+		softLimitStates_.insert(currentAxis, softLimitStates.value(currentAxis));
+	}
+}
+
+bool AMPIC887HexapodState::limitSwitchState(AMGCS2::Axis axis) const
+{
+	return limitSwitchStates_.value(axis, false);
+}
+
+const AMPIC887AxisMap<bool> AMPIC887HexapodState::limitSwitchStates() const
+{
+	return limitSwitchStates_;
+}
+
+AMGCS2::PositionUnits AMPIC887HexapodState::positionUnits(AMGCS2::Axis axis) const
+{
+	return positionUnits_.value(axis, AMGCS2::UnknownPositionUnit);
+}
+
+const AMPIC887AxisMap<AMGCS2::PositionUnits> AMPIC887HexapodState::positionUnits()
+{
+	return positionUnits_;
+}
+
+double AMPIC887HexapodState::stepSize(AMGCS2::Axis axis) const
+{
+	return stepSizes_.value(axis, 0);
+}
+
+const AMPIC887AxisMap<double> AMPIC887HexapodState::stepSizes() const
+{
+	return stepSizes_;
+}
+
+void AMPIC887HexapodState::setStepSize(AMGCS2::Axis axis, double stepSize)
+{
+	if(axis == AMGCS2::UnknownAxis) {
+		return;
+	}
+
+	stepSizes_.insert(axis, stepSize);
+}
+
+void AMPIC887HexapodState::setStepSizes(const AMPIC887AxisMap<double> &stepSizes)
+{
+	AMPIC887AxisCollection axes = stepSizes.axes();
+	foreach(AMGCS2::Axis currentAxis, axes) {
+
+		stepSizes_.insert(currentAxis, stepSizes.value(currentAxis));
+	}
+}
+
+double AMPIC887HexapodState::minCommandablePosition(AMGCS2::Axis axis) const
+{
+	return minCommandablePositions_.value(axis, 0);
+}
+
+const AMPIC887AxisMap<double> AMPIC887HexapodState::minCommandablePositions() const
+{
+	return minCommandablePositions_;
+}
+
+double AMPIC887HexapodState::maxCommandablePosition(AMGCS2::Axis axis) const
+{
+	return maxCommandablePositions_.value(axis, 0);
+}
+
+const AMPIC887AxisMap<double> AMPIC887HexapodState::maxCommandablePositions() const
+{
+	return maxCommandablePositions_;
+}
+
+double AMPIC887HexapodState::pivotPoint(AMGCS2::Axis axis) const
+{
+	return pivotPoints_.value(axis, 0);
+}
+
+const AMPIC887AxisMap<double> AMPIC887HexapodState::pivotPoints() const
+{
+	return pivotPoints_;
+}
+
+void AMPIC887HexapodState::setPivotPoint(AMGCS2::Axis axis, double pivotPoint)
+{
+	if(axis == AMGCS2::UnknownAxis ||
+			axis == AMGCS2::UAxis ||
+			axis == AMGCS2::VAxis ||
+			axis == AMGCS2::WAxis) {
+		return;
+	}
+
+	pivotPoints_.insert(axis, pivotPoint);
```

Comment:
Weren't there more rules for setting pivot points?  Something like u, v, and w all had to be 0 as well?